libpdbg/device.c

```diff
@@ -703,13 +764,13 @@ static void pdbg_targets_init_virtual(struct pdbg_target *node, struct pdbg_targ
 
 bool pdbg_targets_init(void *fdt)
 {
-	struct pdbg_dtb *dtb;
-
 	if (pdbg_dt_root) {
-		PR_WARNING("pdbg_targets_init() must be called only once\n");
-		return true;
+		PR_INFO("pdbg_dt_root is already pointing to a dev tree, resetting it for the new one\n");
+		pdbg_reset_dt_root();
```

Comment:
i think this one should not be embedded here rather called separately, we do this only when we want to swtich, that way no change is required for pdbg_set_backend method

Reply:
> i think this one should not be embedded here rather called separately, we do this only when we want to swtich, that way no change is required for pdbg_set_backend method, also clearly states we are calling pdbg_reset_dt_root as we want to switch backend.

My idea was also that but @dhruvibm and @aravynd suggested that it shouldn't be available as an API for user as it is a performance intensive call and should be only used as an internal calls. @dhruvibm and @aravynd can you please confirm once again?

Reply:
Yeah thats what we thought.